graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/toFederatedSchema.kt

```diff
@@ -38,5 +38,8 @@ fun toFederatedSchema(
     subscriptions: List<TopLevelObject> = emptyList()
 ): GraphQLSchema {
     val generator = FederatedSchemaGenerator(config)
-    return generator.generate(queries, mutations, subscriptions)
+    generator.addExtendedTypes()
```

Comment:
I believe this should be updated to move to `generateSchema` within the `FederatedSchemaGenerator`?

Reply:
This one I can not move easily with our testing strategy as is.

All the tests in `com.expediagroup.graphql.federation.validation` build a schema to test the `FederatedSchemaValidator` if we add all the types with `@extends` then the schema is no longer valid and it errors out.

Ideally we would not have to call `FederatedSchemaGenerator.generateSchema` to test the validator but that requires changing all the test models to graphql-java fully built schema objects which is a lot more work.

Reply:
If thats the case then I'd drop the `FederatedSchemaGenerator` and make `addExtendedTypes` extension method on the regular `SchemaGenerator`

Reply:
I can't do that because the `ExtendsDirective` is only part of the `graphql-kotlin-federation` package, so there needs to be code there that can reference that annotation. That is why I atleast made the `addAdditionalTypesWithAnnotation` part of the base `SchemaGenerator`

Reply:
I meant the extension should be within the `graphql-kotlin-federation` package (potentially within the same `toFederatedSchema` file.

Reply:
I could do that. It would require though that we expose the `addAdditionalTypesWithAnnotation` as public so we can call it from the extension method. Do we want to to that?

Reply:
so going back to the previous comment -> if I call `FederatedSchemaGenerator.generateSchema` I would expect it to build me a federated schema. With this change it won't be true and requires one to explicitly call the `addExtendedTypes` BEFORE calling `generateSchema`. I understand the reason for that was to make it more testable but it breaks the logic.

Reply:
I agree that it would makes more sense, so how about I make the change to make the testing easier in a separate PR and then we can come back to this one.

Reply:
@dariuszkuc With this change, I will be able to move the addExtendedTypes into the generateSchema function

https://github.com/ExpediaGroup/graphql-kotlin/pull/540

Reply:
Done, this is now in the `generateSchema` method